.github/workflows/build-python-pdoc.yml

```diff
@@ -0,0 +1,78 @@
+name: DiskANN Build PDoc Documentation
+on: [workflow_call]
```

Comment:
I think we want the trigger for this to be on workflow call and on push to main, and then remove it from the push test entirely.

Otherwise competing branch pushes could maybe end up doing undesired things if they both try to publish the docs at the same time (even ignoring the race conditions, now people pushing updates need to hope nobody else who *isn't* pushing updates to their branches aren't pushing, else they won't see their changes reflected.

So we should have something like:

```yaml
on:
  workflow_call:
  push:
    branches:
      - 'main'
```
And then remove it from the `push-test-yml` entirely.

Reply:
Nevermind all of this, I see now you have the branch checks to keep it from publishing, instead people just download the documentation artifact to validate.  That works better.